### PR TITLE
EG-474 2i review list data

### PR DESCRIPTION
**Pull request by LesWilson**
Implementation of approval/:id/2i-review end point
Controller calls dummy service that returns a canned data response
New Error object created StaleDataError and returned when process not in correct state to be 2i reviewed

### CODE REVIEW

File: app/services/ApprovalService.scala [hunk 1]
```diff
@@ -66,4 +68,22 @@ class ApprovalService @Inject() (repository: ApprovalRepository) {
...
+        List(PageReview("id1", "how-did-you-earn-extra-income", "NotStarted"),
+          PageReview("id2", "sold-goods-or-services/did-you-only-sell-personal-possessions", "NotStarted"),
+          PageReview("id3", "sold-goods-or-services/have-you-made-a-profit-of-6000-or-more", "NotStarted"),
+          PageReview("id4", "sold-goods-or-services/have-you-made-1000-or-more", "NotStarted"),
+          PageReview("id5", "sold-goods-or-services/you-do-not-need-to-tell-hmrc", "NotStarted"),
+          PageReview("id6", "rent-a-property/do-you-receive-any-income", "NotStarted"),
+          PageReview("id7", "rent-a-property/have-you-rented-out-a-room", "NotStarted")))
+    }
+    Future(Right(Json.toJson(mockData()).as[JsObject]))
+  }
```
Comment (macgrewal) [inline app/services/ApprovalService.scala:87]: Can this new function be moved into a `ReviewService`?

File: app/models/ApprovalProcessReview.scala [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+object ApprovalProcessReview {
+  implicit val formats: OFormat[ApprovalProcessReview] = Json.format[ApprovalProcessReview]
+}
+
+case class PageReview(id: String, title: String, status: String)
+
+object PageReview {
+  implicit val formats: OFormat[PageReview] = Json.format[PageReview]
+}
```
Comment (macgrewal) [inline app/models/ApprovalProcessReview.scala:33]: Please move into a `PageReview.scala` file as it makes finding classes easier when not using an IDE e.g. terminal, GitHub UI or file browser.

Also, this is recommended in the [Scala Documentation](https://docs.scala-lang.org/style/files.html) site

Comment (LesWilson) [reply app/models/ApprovalProcessReview.scala:33]: Done - will be in next push

File: app/services/ApprovalService.scala (hunk 1 above)
Comment (LesWilson) [reply app/services/ApprovalService.scala:87]: Done - will be in next push

File: app/controllers/ApprovalController.scala [hunk 3]
```diff
@@ -54,4 +55,13 @@ class ApprovalController @Inject() (approvalService: ApprovalService, cc: Contro
...
 
+  def approval2iReviewInfo(id: String): Action[AnyContent] = Action.async { _ =>
+    approvalService.approval2iReviewInfo(id).map {
+      case Right(data) => Ok(data)
+      case Left(Errors(NotFoundError :: Nil)) => NotFound(Json.toJson(NotFoundError))
+      case Left(Errors(StaleDataError :: Nil)) => NotFound(Json.toJson(StaleDataError))
+      case Left(Errors(BadRequestError :: Nil)) => BadRequest(Json.toJson(BadRequestError))
+      case Left(_) => InternalServerError(Json.toJson(InternalServiceError))
+    }
+  }
```
[the remark this answers is not in the document]
Comment (LesWilson) [reply app/controllers/ApprovalController.scala:66]: Done - will be in next push

File: app/services/ApprovalService.scala [hunk 4]
```diff
@@ -66,4 +68,22 @@ class ApprovalService @Inject() (repository: ApprovalRepository) {
...
+        LocalDate.of(2020, 5, 10),
+        List(PageReview("id1", "how-did-you-earn-extra-income", "NotStarted"),
+          PageReview("id2", "sold-goods-or-services/did-you-only-sell-personal-possessions", "NotStarted"),
+          PageReview("id3", "sold-goods-or-services/have-you-made-a-profit-of-6000-or-more", "NotStarted"),
+          PageReview("id4", "sold-goods-or-services/have-you-made-1000-or-more", "NotStarted"),
+          PageReview("id5", "sold-goods-or-services/you-do-not-need-to-tell-hmrc", "NotStarted"),
+          PageReview("id6", "rent-a-property/do-you-receive-any-income", "NotStarted"),
+          PageReview("id7", "rent-a-property/have-you-rented-out-a-room", "NotStarted")))
+    }
+    Future(Right(Json.toJson(mockData()).as[JsObject]))
```
[the remark this answers is not in the document]
Comment (LesWilson) [reply app/services/ApprovalService.scala:86]: Done - will be in next push

File: app/services/ApprovalService.scala [hunk 5]
```diff
@@ -66,4 +68,22 @@ class ApprovalService @Inject() (repository: ApprovalRepository) {
       case Right(success) => Right(Json.toJson(success).as[JsArray])
     }
   }
+
+  def approval2iReviewInfo(id: String): Future[RequestOutcome[JsObject]] = {
```
[the remark this answers is not in the document]
Comment (LesWilson) [reply app/services/ApprovalService.scala:72]: Done - will be in next push

[the remark this answers is not in the document]
Comment (LesWilson) [reply app/services/ApprovalService.scala:72]: Done - will be in next push

**Pull request merged**